node/src/actors/session/handlers.rs

```diff
@@ -254,18 +254,30 @@ impl StreamHandler<BytesMut, Error> for Session {
                         SessionStatus::Consolidated,
                         Command::LastBeacon(LastBeacon {
                             highest_block_checkpoint,
+                            highest_superblock_checkpoint,
                         }),
                     ) => {
-                        session_last_beacon_inbound(self, ctx, highest_block_checkpoint);
+                        session_last_beacon_inbound(
+                            self,
+                            ctx,
+                            highest_block_checkpoint,
+                            highest_superblock_checkpoint,
```

Comment:
This looks redundant, `highest_block_checkpoint == highest_superblock_checkpoint`, right?

Reply:
Maybe you can change

                        Command::LastBeacon(LastBeacon {
                            highest_block_checkpoint,
                            highest_block_checkpoint,
                            highest_superblock_checkpoint,
                        }),

Into

    Command::LastBeacon(last_beacon)

And pass the two beacons at once:

```suggestion
                            last_beacon
```